hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java

```diff
@@ -179,6 +212,104 @@ public void run() {
         // top-level command, do nothing
     }
 
+    @Command(description = "Starts the SQL shell [BETA]")
+    public void sql(@Mixin(name = "verbosity") Verbosity verbosity,
+                    @Mixin(name = "targets") TargetsMixin targets
+    ) {
+        runWithJet(targets, verbosity, true, jet -> {
+            LineReader reader = LineReaderBuilder.builder().parser(new MultilineParser())
+                    .variable(LineReader.SECONDARY_PROMPT_PATTERN, new AttributedStringBuilder()
+                            .style(AttributedStyle.BOLD.foreground(SECONDARY_COLOR)).append("%M%P > ").toAnsi())
+                    .variable(LineReader.INDENTATION, 2)
+                    .option(LineReader.Option.DISABLE_EVENT_EXPANSION, true)
+                    .appName("hazelcast-jet-sql")
+                    .build();
+
+            AtomicReference<SqlResult> activeSqlResult = new AtomicReference<>();
+            reader.getTerminal().handle(Signal.INT, signal -> {
+                SqlResult r = activeSqlResult.get();
+                if (r != null) {
+                    r.close();
+                }
+            });
+
+            PrintWriter writer = reader.getTerminal().writer();
+            writer.println(sqlStartingPrompt(jet));
+            writer.flush();
+
+            for (; ; ) {
+                String command;
+                try {
+                    command = reader.readLine(new AttributedStringBuilder()
+                            .style(AttributedStyle.DEFAULT.foreground(SECONDARY_COLOR))
+                            .append("sql> ").toAnsi()).trim();
+                } catch (EndOfFileException | IOError e) {
+                    // Ctrl+D, and kill signals result in exit
+                    writer.println(SQLCliConstants.EXIT_PROMPT);
+                    writer.flush();
+                    break;
+                } catch (UserInterruptException e) {
+                    // Ctrl+C cancels the not-yet-submitted query
+                    continue;
+                }
+
+                command = command.trim();
+                if (command.length() > 0 && command.lastIndexOf(";") == (command.length() - 1)) {
```

Comment:
The condition can be simplified:
```suggestion
                if (command.length() > 0 && command.charAt(command.length() - 1) == ';') {
```

But the issue is that this condition breaks the `;` handling in `MultilineParser`. This command failse (note there's no `;` at the end):
```
sql> select ';' from foo
There are non-whitespace characters after the semicolon.
```